app/src/main/java/com/example/lastfm/MainActivity.java

```diff
@@ -4,81 +4,128 @@
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
 import java.util.Arrays;
 import java.util.Collection;
 
 public class MainActivity extends AppCompatActivity {
-    private ImageView newsImage;
-    private TextView textHeader;
-    private TextView textDescription;
 
 
-    private RecyclerView newsRecyclerView;
-    private NewsAdapter newsAdapter;
 
+    private static final String APP_ID = "b4ab3bf82dcb495e182e04cfc1f12b7b";
+
+
+    private NewsAdapter newsAdapter;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        String url = String.format("http://ws.audioscrobbler.com/2.0/?method=chart.gettopartists&limit=1&api_key=b4ab3bf82dcb495e182e04cfc1f12b7b&format=json");
 
         initRecyclerView();
+        loadArtists();
+        TextView textView = findViewById(R.id.asynctask_view);
+        new GetTopTask(textView).execute(url);
+    }
+
+    private static class GetTopTask extends AsyncTask<String, Void, String> {
+        private TextView tvTop;
+
+        GetTopTask(TextView textView) {
+            this.tvTop = textView;
+        }
+        @Override
+        protected String doInBackground(String... strings) {
+            String artistName = "UNDEFINED";
+            try {
+                URL url = new URL(strings[0]);
+                HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
+
+                InputStream stream = new BufferedInputStream(urlConnection.getInputStream());
+                BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(stream));
+                StringBuilder builder = new StringBuilder();
+
+                String inputString;
+                while ((inputString = bufferedReader.readLine()) != null) {
+                    builder.append(inputString);
+                }
+
+                JSONObject topLevel = new JSONObject(builder.toString());
+                JSONObject main = topLevel.getJSONObject("artists");
+                System.out.println(main);
+                JSONArray artist = (JSONArray) main.get("artist");
+                System.out.println(artist);
+
+
+                urlConnection.disconnect();
+            } catch (IOException | JSONException e) {
+                e.printStackTrace();
+            }
+            return artistName;
+        }
 
-        loadNews();
+        @Override
+        protected void onPostExecute(String artistName) {
+            tvTop.setText("Top artist:" + artistName);
+        }
     }
 
-        private void loadNews(){
-            Collection<NewsData> news = getNews();
+        private void loadArtists(){
+            Collection<NewsData> news = getArtists();
             newsAdapter.setItems(news);
         }
 
-
-        private Collection<NewsData> getNews(){
+        private Collection<NewsData> getArtists(){
```

Comment:
the method "getArtists" must not return a collection of objects NewsData. And it's better to use List instead of Collection

Reply:
Changed to ArrayList

Reply:
u didn't change the returned type

Reply:
This string is deleted.